File: Android-Demo/EncryptionDemo/app/src/main/java/com/wx/encryptiondemo/base/BaseActivity.java
package com.wx.encryptiondemo.base;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity{

    public abstract int getLayoutId();

    public abstract void setDataBinding(ViewDataBinding binding);

    public abstract void initData();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setDataBinding(DataBindingUtil.setContentView(this,getLayoutId()));
        initData();
    }

    public void openActivity(Class<? extends BaseActivity> otherActivity) {
        Intent intent = new Intent(this,otherActivity);
        startActivity(intent);
    }

}

File: 15-18/BaseMaster1/app/src/main/java/com/wx/basemaster1/base/BaseActivity.kt
package com.wx.basemaster1.base

import android.app.Activity
import android.content.Intent
import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.os.Bundle
import android.support.annotation.LayoutRes

abstract class BaseActivity : Activity() {

    @get:LayoutRes
    abstract val layoutId: Int

    abstract fun setDataBinding(binding: ViewDataBinding)

    abstract fun initData()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setDataBinding(DataBindingUtil.setContentView<ViewDataBinding>(this, layoutId))
        initData()
    }

    fun openActivity(other: Class<out Activity>) {
        val intent = Intent(this, other)
        startActivity(intent)
    }

}
